### PR TITLE
Use np.interp instead of interpolation.interp

### PR DESCRIPTION
**Pull request by kp992**
This PR replaces all the occurences of `interpolation.interp` to `np.interp` now that `numba` supports `np.interp` directly.

### CODE REVIEW

**Comment by jstac** [comment]
@mmcky and @HumphreyYang , could you please do a first round review of this?

The aim is that swapping to NumPy's interp does not change results or degrade performance.

**Comment by jstac** [comment]
Note that this is related to https://github.com/QuantEcon/lecture-python.myst/pull/387

File: lectures/BCG_incomplete_mkts.md [hunk 1]
```diff
@@ -1943,3 +1942,7 @@ Agents of type 2 value bonds more highly (they want more hedging).
 
 Taken together with our earlier plot of equity holdings, these graphs confirm our earlier conjecture that while both type
 of agents hold equities, only agents of type 2 holds bonds.
+
+```{code-cell} ipython3
```
Comment (mmcky) [inline lectures/BCG_incomplete_mkts.md:1946]:
```suggestion
```

File: lectures/BCG_incomplete_mkts.md [hunk 2]
```diff
@@ -1943,3 +1942,7 @@ Agents of type 2 value bonds more highly (they want more hedging).
...
+
+```
```
Comment (mmcky) [inline lectures/BCG_incomplete_mkts.md:1948]:
```suggestion
```

File: lectures/coase.md [hunk 3]
```diff
@@ -24,14 +26,7 @@ kernelspec:
...
 
-In addition to what's in Anaconda, this lecture will need the following libraries:
-
-```{code-cell} ipython
----
-tags: [hide-output]
----
-!pip install interpolation
-```
++++
```
Comment (mmcky) [inline lectures/coase.md:29]:
```suggestion
```

File: lectures/lucas_model.md [hunk 4]
```diff
@@ -27,14 +29,7 @@ kernelspec:
...
 
-In addition to what's in Anaconda, this lecture will need the following libraries:
-
-```{code-cell} ipython
----
-tags: [hide-output]
----
-!pip install interpolation
-```
++++
```
Comment (mmcky) [inline lectures/lucas_model.md:32]:
```suggestion
```

**Comment by mmcky** [comment]
Thanks @kp992. 

The last dependency on `interpolations` is in `amss`

```
lectures/amss.md:from interpolation.splines import eval_linear, UCGrid, nodes
```

@kp992 any scope to be able to remove interpolations as a dependency -- otherwise maybe we need to solve https://github.com/EconForge/interpolation.py/issues/110 re: `generated_jit` to make use of this package in `amss`

File: lectures/BCG_incomplete_mkts.md (hunk 1 above)
Comment (kp992) [reply lectures/BCG_incomplete_mkts.md]: Done

**Comment by kp992** [comment]
Thanks for the review @mmcky.

> The last dependency on interpolations is in amss

Yeah, I checked that. Its the only one left with `interpolation.py`.




> @kp992 any scope to be able to remove interpolations as a dependency -- otherwise maybe we need to solve [EconForge/interpolation.py#110](https://github.com/EconForge/interpolation.py/issues/110) re: `generated_jit` to make use of this package in `amss`

Hmm, I will check with some scipy functions and if possible remove it in a follow-up PR.

**Comment by mmcky** [comment]
@kp992 the lecture `opt_tax_recur` is failing due to issues with the `nelder_mead` function from `quantecon`

```
Traceback (most recent call last):
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
log_example = LogUtility()
# Solve sequential problem
seq_log = SequentialLS(log_example)

# Initialize grid for value function iteration and solve
x_grid = np.linspace(-3., 3., 200)

# Solve recursive problem
rec_log = RecursiveLS(log_example, x_grid)

T_length = 20
sHist = np.array([0, 0, 0, 0, 0,
                  0, 0, 0, 1, 1,
                  0, 0, 0, 1, 1,
                  1, 1, 1, 1, 0])

# Simulate
sim_seq = seq_log.simulate(0.5, 0, T_length, sHist)
sim_rec = rec_log.simulate(0.5, 0, T_length, sHist)

fig, axes = plt.subplots(3, 2, figsize=(14, 10))
titles = ['Consumption', 'Labor Supply', 'Government Debt',
          'Tax Rate', 'Government Spending', 'Output']

for ax, title, sim_s, sim_b in zip(axes.flatten(), titles, sim_seq[:6], sim_rec[:6]):
                                   ax.plot(sim_s, '-ob', sim_b, '-xk', alpha=0.7)
                                   ax.set(title=title)
                                   ax.grid()

axes.flatten()[0].legend(('Sequential', 'Recursive'))
fig.tight_layout()
plt.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mTypingError[0m                               Traceback (most recent call last)
Cell [0;32mIn[11], line 19[0m
[1;32m     17[0m [38;5;66;03m# Simulate[39;00m
[1;32m     18[0m sim_seq [38;5;241m=[39m seq_log[38;5;241m.[39msimulate([38;5;241m0.5[39m, [38;5;241m0[39m, T_length, sHist)
[0;32m---> 19[0m sim_rec [38;5;241m=[39m rec_log[38;5;241m.[39msimulate([38;5;241m0.5[39m, [38;5;241m0[39m, T_length, sHist)
[1;32m     21[0m fig, axes [38;5;241m=[39m plt[38;5;241m.[39msubplots([38;5;241m3[39m, [38;5;241m2[39m, figsize[38;5;241m=[39m([38;5;241m14[39m, [38;5;241m10[39m))
[1;32m     22[0m titles [38;5;241m=[39m [[38;5;124m'[39m[38;5;124mConsumption[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mLabor Supply[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mGovernment Debt[39m[38;5;124m'[39m,
[1;32m     23[0m           [38;5;124m'[39m[38;5;124mTax Rate[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mGovernment Spending[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mOutput[39m[38;5;124m'[39m]

Cell [0;32mIn[4], line 126[0m, in [0;36mRecursiveLS.simulate[0;34m(self, b0, s0, T, sHist)[0m
[1;32m    123[0m RHist [38;5;241m=[39m np[38;5;241m.[39mzeros(T[38;5;241m-[39m[38;5;241m1[39m)
[1;32m    125[0m [38;5;66;03m# Time 0[39;00m
[0;32m--> 126[0m [38;5;28mself[39m[38;5;241m.[39mtime0_allocation(b0, s0)
[1;32m    127[0m cHist[[38;5;241m0[39m], nHist[[38;5;241m0[39m], xHist[[38;5;241m0[39m] [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mc0, [38;5;28mself[39m[38;5;241m.[39mn0, [38;5;28mself[39m[38;5;241m.[39mxprime0[s0]
[1;32m    128[0m τHist[[38;5;241m0[39m] [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mτ(cHist[[38;5;241m0[39m], nHist[[38;5;241m0[39m])

Cell [0;32mIn[4], line 79[0m, in [0;36mRecursiveLS.time0_allocation[0;34m(self, b0, s0)[0m
[1;32m     74[0m [38;5;250m[39m[38;5;124;03m'''[39;00m
[1;32m     75[0m [38;5;124;03mFind the optimal time 0 allocation by maximization.[39;00m
[1;32m     76[0m [38;5;124;03m'''[39;00m
[1;32m     78[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mV [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[0;32m---> 79[0m     [38;5;28mself[39m[38;5;241m.[39mtime1_allocation()
[1;32m     81[0m π, g, S [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mπ, [38;5;28mself[39m[38;5;241m.[39mg, [38;5;28mself[39m[38;5;241m.[39mS
[1;32m     82[0m pref, x_grid, bounds [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mpref, [38;5;28mself[39m[38;5;241m.[39mx_grid, [38;5;28mself[39m[38;5;241m.[39mbounds

Cell [0;32mIn[4], line 57[0m, in [0;36mRecursiveLS.time1_allocation[0;34m(self, V, tol)[0m
[1;32m     53[0m         z[:, s, i[38;5;241m+[39m[38;5;241m2[39m] [38;5;241m=[39m x_grid
[1;32m     55[0m [38;5;28;01mwhile[39;00m [38;5;28;01mTrue[39;00m:
[1;32m     56[0m     [38;5;66;03m# value function iteration[39;00m
[0;32m---> 57[0m     V_new, z_new [38;5;241m=[39m T(V, z, pref, π, g, x_grid, bounds)
[1;32m     59[0m     [38;5;28;01mif[39;00m np[38;5;241m.[39mmax(np[38;5;241m.[39mabs(V [38;5;241m-[39m V_new)) [38;5;241m<[39m tol:
[1;32m     60[0m         [38;5;28;01mbreak[39;00m

File [0;32m~/miniconda3/envs/quantecon/lib/python3.11/site-packages/numba/core/dispatcher.py:468[0m, in [0;36m_DispatcherBase._compile_for_args[0;34m(self, *args, **kws)[0m
[1;32m    464[0m         msg [38;5;241m=[39m ([38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00m[38;5;28mstr[39m(e)[38;5;241m.[39mrstrip()[38;5;132;01m}[39;00m[38;5;124m [39m[38;5;130;01m\n[39;00m[38;5;130;01m\n[39;00m[38;5;124mThis error may have been caused [39m[38;5;124m"[39m
[1;32m    465[0m                [38;5;124mf[39m[38;5;124m"[39m[38;5;124mby the following argument(s):[39m[38;5;130;01m\n[39;00m[38;5;132;01m{[39;00margs_str[38;5;132;01m}[39;00m[38;5;130;01m\n[39;00m[38;5;124m"[39m)
[1;32m    466[0m         e[38;5;241m.[39mpatch_message(msg)
[0;32m--> 468[0m     error_rewrite(e, [38;5;124m'[39m[38;5;124mtyping[39m[38;5;124m'[39m)
[1;32m    469[0m [38;5;28;01mexcept[39;00m errors[38;5;241m.[39mUnsupportedError [38;5;28;01mas[39;00m e:
[1;32m    470[0m     [38;5;66;03m# Something unsupported is present in the user code, add help info[39;00m
[1;32m    471[0m     error_rewrite(e, [38;5;124m'[39m[38;5;124munsupported_error[39m[38;5;124m'[39m)

File [0;32m~/miniconda3/envs/quantecon/lib/python3.11/site-packages/numba/core/dispatcher.py:409[0m, in [0;36m_DispatcherBase._compile_for_args.<locals>.error_rewrite[0;34m(e, issue_type)[0m
[1;32m    407[0m     [38;5;28;01mraise[39;00m e
[1;32m    408[0m [38;5;28;01melse[39;00m:
[0;32m--> 409[0m     [38;5;28;01mraise[39;00m e[38;5;241m.[39mwith_traceback([38;5;28;01mNone[39;00m)

[0;31mTypingError[0m: Failed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1mNameError: name 'interp' is not defined[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function obj_V at 0x7f3021f1fba0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (443)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function obj_V at 0x7f3021f1fba0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (443)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 443:[0m
[1mdef _neg_bounded_fun(fun, bounds, x, args=()):
    <source elided>
    if _check_bounds(x, bounds):
[1m        return -fun(x, *args)
[0m        [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (226)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (238)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (258)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (269)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (281)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (226)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (238)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (258)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (269)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (281)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 196:[0m
[1mdef _nelder_mead_algorithm(fun, vertices, bounds=np.array([[], []]).T,
    <source elided>
    for i in range(n+1):
[1m        f_val[i] = _neg_bounded_fun(fun, bounds, vertices[i], args=args)
[0m        [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _nelder_mead_algorithm at 0x7f30232c4680>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (118)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _nelder_mead_algorithm at 0x7f30232c4680>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (118)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 118:[0m
[1m    ... def rosenbrock(x):
        <source elided>

[1m    results = _nelder_mead_algorithm(fun, vertices, bounds, args=args,
[0m    [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function nelder_mead at 0x7f30232c4220>))[0m
[0m[1mDuring: typing of call at /tmp/ipykernel_5804/222986336.py (173)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function nelder_mead at 0x7f30232c4220>))[0m
[0m[1mDuring: typing of call at /tmp/ipykernel_5804/222986336.py (173)
[0m
[1m
File "../../../../../../tmp/ipykernel_5804/222986336.py", line 173:[0m
[1m<source missing, REPL/exec in use?>[0m

TypingError: Failed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1m[1m[1m[1mFailed in nopython mode pipeline (step: nopython frontend)
[1mNameError: name 'interp' is not defined[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function obj_V at 0x7f3021f1fba0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (443)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function obj_V at 0x7f3021f1fba0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (443)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 443:[0m
[1mdef _neg_bounded_fun(fun, bounds, x, args=()):
    <source elided>
    if _check_bounds(x, bounds):
[1m        return -fun(x, *args)
[0m        [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (226)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (238)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (258)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (269)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (281)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (226)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (238)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (258)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (269)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (281)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _neg_bounded_fun at 0x7f30232c4fe0>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (196)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 196:[0m
[1mdef _nelder_mead_algorithm(fun, vertices, bounds=np.array([[], []]).T,
    <source elided>
    for i in range(n+1):
[1m        f_val[i] = _neg_bounded_fun(fun, bounds, vertices[i], args=args)
[0m        [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _nelder_mead_algorithm at 0x7f30232c4680>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (118)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function _nelder_mead_algorithm at 0x7f30232c4680>))[0m
[0m[1mDuring: typing of call at /home/runner/miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py (118)
[0m
[1m
File "../../../../miniconda3/envs/quantecon/lib/python3.11/site-packages/quantecon/optimize/nelder_mead.py", line 118:[0m
[1m    ... def rosenbrock(x):
        <source elided>

[1m    results = _nelder_mead_algorithm(fun, vertices, bounds, args=args,
[0m    [1m^[0m[0m

[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function nelder_mead at 0x7f30232c4220>))[0m
[0m[1mDuring: typing of call at /tmp/ipykernel_5804/222986336.py (173)
[0m
[0m[1mDuring: resolving callee type: type(CPUDispatcher(<function nelder_mead at 0x7f30232c4220>))[0m
[0m[1mDuring: typing of call at /tmp/ipykernel_5804/222986336.py (173)
[0m
[1m
File "../../../../../../tmp/ipykernel_5804/222986336.py", line 173:[0m
[1m<source missing, REPL/exec in use?>[0m
```

**Comment by kp992** [comment]
@mmcky Thanks for pointing out the issue. The CI is green.

**Comment by mmcky** [comment]
thanks @kp992 

Just documenting this is now the only use case of `interpolations` left

```
lectures/amss.md:from interpolation.splines import eval_linear, UCGrid, nodes
```

**Pull request merged**